src/kili/adapters/kili_api_gateway/helpers/queries.py

```diff
@@ -86,10 +95,19 @@ def execute_query_from_paginated_call(
                     if len(elements) == 0:
                         break
 
-                    yield from elements
+                    if unicity_field is None:
+                        yield from elements
+                        pbar.update(len(elements))
+                    else:
+                        for element in elements:
+                            unicity_value = element[unicity_field]
+
+                            if unicity_value not in unicity_values:
```

Comment:
else, shouldn't we increase the `count_elements_retrieved` ?

Reply:
In case of unicity check, `count_elements_retrieved` is only used to determine the next elements to retrieve so it should not depend on the duplicates.

When unicity check is enabled, we retrieve elements until there is no more to get if `first` property is not provided, otherwise we stop when the number of unique elements reach `first` value.